aten/src/ATen/FunctionalTensorWrapper.cpp

```diff
@@ -249,6 +253,10 @@ TensorList to_functional_tensor(const TensorList& t_list) {
 }
 
 Tensor from_functional_tensor(const Tensor& tensor) {
+  // Note [Wrapped Numbers <> Functionalization]
+  if (tensor.unsafeGetTensorImpl()->is_wrapped_number()) {
+      return tensor;
+  }
```

Comment:
What happens if one performs operations between FunctionalTensorWrapper and non-FunctionalTensorWrapper?

Reply:
Answered my question: if things go through the fallback, then the fallback "unwraps everything that is not a wrapped number" via from_functional_tensor, calls the op, and then re-wraps with to_functional_tensor. The generated code for in-place operations and views assumes that wrapped numbers don't occur

Reply:
yep, that's right. For functorch, this also means that mixed functionalized/non-functionalized tensors currently don't work; something like:
```
y = torch.ones(2)
def f(x):
   return x + y
out = functionalize(f)(torch.ones(2))
```